### ISSUE TITLE
no need for the reference blas

### ISSUE THREAD

**Issue by fommil** [haifengl/smile#207]
Your netlib-java dependencies are too large, you only need to depend on the java and system variants. The "reference" versions are a waste of download space.

Also, I'd appreciate a link from your documentation to https://github.com/fommil/netlib-java#netlib-java when you refer to native binaries as I spent a lot of time on this project and have not been remunerated, hence technical credit is a substitute currency.

**Comment by haifengl** [comment]
Thanks! Your github README.md says

```
"com.github.fommil.netlib" % "all" % "1.1.2" pomOnly()
```
for dependency. Can you please point me to more detailed information of the best practice to add netlib-java dependency? I am not clear "only need to depend on the java and system variants. The "reference" versions are a waste of download space." Thanks again!

We surely will add your project link to all relevant pages in our documents. Thanks for the great work!

**Comment by fommil** [comment]
Just look it the pom you're depending on and only pick the bits you need. You probably don't want the raspberry pi binaries either.

**Comment by benmccann** [comment]
So perhaps replace it with the following?

```
"com.github.fommil.netlib" % "netlib-native_system-linux-i686" % "1.1"
"com.github.fommil.netlib" % "netlib-native_system-linux-x86_64" % "1.1"
"com.github.fommil.netlib" % "netlib-native_system-win-x86_64" % "1.1"
"com.github.fommil.netlib" % "netlib-native_system-win-i686" % "1.1"
"com.github.fommil.netlib" % "netlib-native_system-osx-x86_64" % "1.1"
"com.github.fommil.netlib" % "native_system-java" % "1.1"
```

**Comment by fommil** [comment]
assuming they are the latest, that is the kind of thing that would be good. You might even want to drop 32 bit linux. Make sure you also have

```
"com.github.fommil.netlib" % "core" % "1.1.2"
"com.github.fommil" % "jniloader" % "1.1"
```

the JVM fallback logic and JNI loader, respectively.

**Comment by haifengl** [comment]
Thank guys! I will try it.

**Comment by haifengl** [comment]
It works well on Mac. I will test it on Windows and Linux soon. Thanks!

**Comment by haifengl** [comment]
@fommil Do you support MKL on windows? It is easy to install it on windows and it does provide excellent performance. Thanks!

**Comment by fommil** [comment]
Did you not watch the conference talk referenced from the README? I think you should as it is highly relevant for what you're doing here.

**Comment by haifengl** [comment]
Sure I did. Let me rephrase my question. I understand that you support MKL on Linux (with alternatives). On Windows, you look for libblas3.dll and liblapack3.dll. Is it possible that you support MKL out of box on Windows (suppose its dlls are in PATH)? Thanks!

**Comment by haifengl** [comment]
I draw the conclusion too quickly. Check the log again on Mac, I found the following warnings:

```
Aug 18, 2017 10:34:51 PM com.github.fommil.netlib.LAPACK <clinit>
WARNING: Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
Aug 18, 2017 10:34:51 PM com.github.fommil.netlib.LAPACK <clinit>
WARNING: Failed to load implementation from: com.github.fommil.netlib.NativeRefLAPACK
```

Here is my dependencies:

```
libraryDependencies += "net.sourceforge.f2j" % "arpack_combined_all" % "0.1"

libraryDependencies ++= Seq(
  "com.github.fommil.netlib" % "netlib-native_system-linux-x86_64" % "1.1",
  "com.github.fommil.netlib" % "netlib-native_system-win-x86_64"   % "1.1",
  "com.github.fommil.netlib" % "netlib-native_system-osx-x86_64"   % "1.1",
  "com.github.fommil.netlib" % "native_system-java"                % "1.1",
  "com.github.fommil.netlib" % "core"                              % "1.1.2",
  "com.github.fommil"        % "jniloader"                         % "1.1"
)
```

What do I miss? Thanks!

**Comment by fommil** [comment]
Hmm, I would expect that to work.

**Comment by haifengl** [comment]
It seems a bug with SBT (or sbt-native-packager). Although we list libraries like netlib-native_system-osx-x86_64, SBT doesn't include it in the distribution package. I revert the dependency back to `all`.

**Comment by benmccann** [comment]
muuki88 is very responsive in maintaining sbt-native-packager. If you file an issue there with steps to reproduce, he is likely to address it

**Comment by haifengl** [comment]
I submit a ticket:

https://github.com/sbt/sbt-native-packager/issues/1028

Hope it can be fixed in the next 0.13 release. Thanks!
